multi-node/aws/pkg/config/templates/cloud-config-controller

```diff
@@ -437,399 +405,270 @@ write_files:
-                      {
-                        "containerPort": 8080,
-                        "protocol": "TCP"
-                      }
-                    ],
-                    "resources": {
-                      "limits": {
-                        "cpu": "10m",
-                        "memory": "50Mi"
-                      },
-                      "requests": {
-                        "cpu": "10m",
-                        "memory": "50Mi"
-                      }
-                    }
-                  }
-                ],
-                "dnsPolicy": "Default"
-              }
-            }
-          }
-        }
-
-  - path: /srv/kubernetes/manifests/kube-dns-svc.json
+      apiVersion: v1
+      kind: ReplicationController
+      metadata:
+        labels:
+          k8s-app: kube-dns
+          kubernetes.io/cluster-service: 'true'
+          version: v17.1
+        name: kube-dns-v17.1
+        namespace: kube-system
+      spec:
+        replicas: 1
+        selector:
+          k8s-app: kube-dns
+          version: v17.1
+        template:
+          metadata:
+            labels:
+              k8s-app: kube-dns
+              kubernetes.io/cluster-service: 'true'
+              version: v17.1
+          spec:
+            containers:
+            - args:
+              - --domain=cluster.local.
+              - --dns-port=10053
+              image: gcr.io/google_containers/kubedns-amd64:1.5
+              livenessProbe:
+                failureThreshold: 5
+                httpGet:
+                  path: /healthz
+                  port: 8080
+                  scheme: HTTP
+                initialDelaySeconds: 60
+                successThreshold: 1
+                timeoutSeconds: 5
+              name: kubedns
+              ports:
+              - containerPort: 10053
+                name: dns-local
+                protocol: UDP
+              - containerPort: 10053
+                name: dns-tcp-local
+                protocol: TCP
+              readinessProbe:
+                httpGet:
+                  path: /readiness
+                  port: 8081
+                  scheme: HTTP
+                initialDelaySeconds: 30
+                timeoutSeconds: 5
+              resources:
+                limits:
+                  cpu: 100m
+                  memory: 170Mi
+                requests:
+                  cpu: 100m
+                  memory: 70Mi
+            - args:
+              - --cache-size=1000
+              - --no-resolv
+              - --server=127.0.0.1#10053
+              image: gcr.io/google_containers/kube-dnsmasq-amd64:1.3
+              name: dnsmasq
+              ports:
+              - containerPort: 53
+                name: dns
+                protocol: UDP
+              - containerPort: 53
+                name: dns-tcp
+                protocol: TCP
+            - args:
+              - -cmd=nslookup kubernetes.default.svc.cluster.local 127.0.0.1 >/dev/null
```

Comment:
The json -> yaml conversion was done via a tool, so I'm relying on it to make that decision correctly. This won't merge till it passes e2e tests, so that's the other end of this equation.